packages/lib-classifier/src/plugins/tasks/TextTask/models/TextTask.js

```diff
@@ -0,0 +1,27 @@
+import { getRoot, types } from 'mobx-state-tree'
+
+const TextTask = types.model('TextTask', {
+  help: types.optional(types.string, ''),
+  instruction: types.string,
+  required: types.optional(types.boolean, false),
+  taskKey: types.identifier,
+  text_tags: types.array(types.string),
+  type: types.literal('text')
+})
+.views(self => ({
+  get annotation () {
+    const { currentAnnotations } = getRoot(self).classifications
+    if (currentAnnotations && currentAnnotations.size > 0) {
+      return currentAnnotations.get(self.taskKey)
+    }
+    return undefined
+  }
+}))
+.actions(self => ({
+  updateAnnotation(value) {
+    const { addAnnotation } = getRoot(self).classifications
+    addAnnotation(value, self)
```

Comment:
I find it very confusing that we update an annotation by calling `classifications.addAnnotation(value, task)`.